epersongroups.md

```diff
@@ -0,0 +1,310 @@
+# EPerson groups Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**GET /api/eperson/groups**
```

Comment:
We have an eperson category used for epersons and groups. The endpoint stated by @benbosman is what we have currently implemented, the endpoints.md is wrong.
@benbosman can you please fix it as part of this PR?

Reply:
This is already the current endpoint as documented in https://github.com/DSpace/Rest7Contract/blob/master/epersons.md and used in the implementation.

@tdonohue do you agree it's https://github.com/DSpace/Rest7Contract/blob/master/endpoints.md which should be updated?
Also, is there a guideline on which category to use, I didn't see it in the readme

Reply:
I had initially overlooked that this endpoint is already `/api/eperson/groups`. 

As `EPerson` is also a [type of DSpaceObject](https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/eperson/EPerson.java#L46), I don't understand the purpose to creating a separate "eperson" category here for EPerson and Groups. It results in very odd looking request URIs, in my opinion.

If we have a clear reason, then let me know.  Otherwise, my assumption has been that all DSpace objects appear as `/api/core/{model}/` paths, which would imply this should be `/api/core/epersons/`.  That said, this decision isn't a fault of this PR, so we can do that refactor in a separate PR as needed.

The README doesn't have documented categories yet beyond what is in https://github.com/DSpace/Rest7Contract/blob/master/endpoints.md